packages/core/src/attestation/Attestation.ts

```diff
@@ -32,245 +33,268 @@ import { DelegationNode } from '../delegation/DelegationNode.js'
  * An [[Attestation]] can be queried from the chain. It's stored on-chain in a map:
  * * the key is the hash of the corresponding claim;
  * * the value is a tuple ([[CType]] hash, account, id of the Delegation, and revoked flag).
+ *
+ * @packageDocumentation
  */
-export class Attestation implements IAttestation {
-  /**
-   * [STATIC] [ASYNC] Queries the chain for a given attestation, by `claimHash`.
-   *
-   * @param claimHash - The hash of the claim that corresponds to the attestation to query.
-   * @returns A promise containing the [[Attestation]] or null.
-   */
-  public static async query(
-    claimHash: IAttestation['claimHash']
-  ): Promise<Attestation | null> {
-    return query(claimHash)
-  }
 
-  /**
-   * [STATIC] [ASYNC] Revokes an attestation. Also available as an instance method.
-   *
-   * @param claimHash - The hash of the claim that corresponds to the attestation to revoke.
-   * @param maxDepth - The number of levels to walk up the delegation hierarchy until the delegation node is found.
-   * @returns A promise containing the unsigned SubmittableExtrinsic (submittable transaction).
-   */
-  public static async getRevokeTx(
-    claimHash: IRequestForAttestation['rootHash'],
-    maxDepth: number
-  ): Promise<SubmittableExtrinsic> {
-    return getRevokeTx(claimHash, maxDepth)
+/**
+ * Queries the chain for a given attestation, by `claimHash`.
+ *
+ * @param input - The hash of the claim that corresponds to the attestation or the full attestation object to query.
+ * @returns A promise containing the [[Attestation]] or null.
+ */
+export async function query(
+  input: IAttestation['claimHash'] | IAttestation
```

Comment:
yeah, but then again we could just use the query directly here and remove the chain function. (In case we clean the chain functions up)
Not sure what is better. I could agree on trying not to do too much and dropping the support for multiple types for now?!

Reply:
I mean that the API, the chain function, and the original query function all accept only the claimHash as a parameter, and I object to adding support for the IAttestation as a parameter

Reply:
ok, I will not add convenience paramters to functions for now.